src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -469,15 +470,7 @@ GetForeignConstraintCommandsInternal(Oid relationId, int flags)
 
 	List *foreignKeyCommands = NIL;
 
-	/*
-	 * Set search_path to NIL so that all objects outside of pg_catalog will be
-	 * schema-prefixed. pg_catalog will be added automatically when we call
-	 * PushOverrideSearchPath(), since we set addCatalog to true;
-	 */
-	OverrideSearchPath *overridePath = GetOverrideSearchPath(CurrentMemoryContext);
-	overridePath->schemas = NIL;
-	overridePath->addCatalog = true;
-	PushOverrideSearchPath(overridePath);
+	PushAndNullifySearchPath();
```

Comment:
In order to make the APIs compatible and make the memory allocation explicit, I'd suggest we pass `MemoryContext` to `PushAndNullifySearchPath ()` and pass it to `GetOverrideSearchPath `